nuxeo-core/nuxeo-core-api/src/main/java/org/nuxeo/ecm/core/api/CursorService.java

```diff
@@ -136,6 +137,7 @@ public ScrollResult scroll(String scrollId, Function<O, String> idExtractor) {
                     break;
                 } else {
                     O obj = cursorResult.next();
+                    results.add(obj);
```

Comment:
I don't like that we record objects all the time when other scroll users (that just call `getResultIds()`, for example for repository doc id scrolling) don't need it. Instead please add a boolean to the method signature to switch on this recording.